### PR TITLE
refactor(autoware_launch): remove planning_launch, and use tier4_planning_launch instead

### PR DESCRIPTION
**Pull request by takayuki5168**
Signed-off-by: Takayuki Murooka <takayuki5168@gmail.com>

## PR Type

- Refactor

## Related Links

I noticed SysI members that planning_launch will be replaced with tier4_planning_launch.
https://star4.slack.com/archives/C4P0NSMB5/p1672203286603929

## Description

Use tier4_planning_launch instead of planning_launch.
Launcher for planning will be unified (by using tier4_planning_launch) among reference designs with this PR.

Parameters were copied from `awf/autoware_launch:autoware_launch/config/planning`
## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Code follows [coding guidelines][coding-guidelines]
- [x] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR**: Required to pass before the merge.
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

**Comment by takayuki5168** [comment]
I confirmed that planning simulator worked well with this PR

File: autoware_launch/config/planning/scenario_planning/parking/freespace_planner/freespace_planner.param.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /**:
   ros__parameters:
     # -- Node Configurations --
-    planning_algorithm: "astar" # options: astar, rrtstar
+    planning_algorithm: "astar"
```
Comment (TakaHoribe) [inline autoware_launch/config/planning/scenario_planning/parking/freespace_planner/freespace_planner.param.yaml:4]:
```suggestion
    planning_algorithm: "astar"  # options: astar, rrtstar
```

File: autoware_launch/config/planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/behavior_path_planner.param.yaml [hunk 2]
```diff
@@ -1,31 +1,18 @@
...
+    backward_length_buffer_for_end_of_lane: 5.0
     backward_length_buffer_for_end_of_pull_over: 5.0
     backward_length_buffer_for_end_of_pull_out: 5.0
+    minimum_lane_change_length: 12.0
     minimum_pull_over_length: 16.0
-
     refine_goal_search_radius_range: 7.5
-
-    # parameters for turn signal decider
     turn_signal_intersection_search_distance: 30.0
```
Comment (TakaHoribe) [inline autoware_launch/config/planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/behavior_path_planner.param.yaml]:
```suggestion
    # parameters for turn signal decider
    turn_signal_intersection_search_distance: 30.0
```

File: autoware_launch/config/planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/behavior_path_planner.param.yaml [hunk 3]
```diff
@@ -1,31 +1,18 @@
...
     turn_signal_intersection_angle_threshold_deg: 15.0
     turn_signal_minimum_search_distance: 10.0
     turn_signal_search_time: 3.0
     turn_signal_shift_length_threshold: 0.3
-
+    turn_signal_on_swerving: true
     path_interval: 2.0
```
Comment (TakaHoribe) [inline autoware_launch/config/planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/behavior_path_planner.param.yaml:20]:
```suggestion
    path_interval: 2.0

    visualize_maximum_drivable_area: true

    lateral_distance_max_threshold: 2.0
    longitudinal_distance_min_threshold: 3.0
    expected_front_deceleration: -1.0
    expected_rear_deceleration: -1.0

    expected_front_deceleration_for_abort: -1.0
    expected_rear_deceleration_for_abort: -2.0

    rear_vehicle_reaction_time: 2.0
    rear_vehicle_safety_time_margin: 2.0

```

File: autoware_launch/config/planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/side_shift/side_shift.param.yaml [hunk 4]
```diff
@@ -6,4 +6,3 @@
       shifting_lateral_jerk: 0.2
       min_shifting_distance: 5.0
       min_shifting_speed: 5.56
```
Comment (TakaHoribe) [inline autoware_launch/config/planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/side_shift/side_shift.param.yaml:8]:
```suggestion
      min_shifting_speed: 5.56
      shift_request_time_limit: 1.0
```

File: autoware_launch/config/planning/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/detection_area.param.yaml [hunk 5]
```diff
@@ -6,4 +6,3 @@
       dead_line_margin: 5.0
       use_pass_judge_line: false
       state_clear_time: 2.0
```
Comment (TakaHoribe) [inline autoware_launch/config/planning/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/detection_area.param.yaml:8]:
```suggestion
      state_clear_time: 2.0
      shift_request_time_limit: 1.0
```

File: autoware_launch/config/planning/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/stop_line.param.yaml [hunk 6]
```diff
@@ -2,6 +2,5 @@
   ros__parameters:
     stop_line:
       stop_margin: 0.0
+      stop_check_dist: 2.0
       stop_duration_sec: 1.0
```
Comment (TakaHoribe) [inline autoware_launch/config/planning/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/stop_line.param.yaml:6]:
```suggestion
      stop_duration_sec: 1.0
      hold_stop_margin_distance: 2.0
      use_initialization_stop_line_state: true
```

File: autoware_launch/config/planning/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/virtual_traffic_light.param.yaml [hunk 7]
```diff
@@ -4,6 +4,5 @@
       max_delay_sec: 3.0
       near_line_distance: 1.0
       dead_line_margin: 1.0
```
Comment (TakaHoribe) [inline autoware_launch/config/planning/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/virtual_traffic_light.param.yaml:6]:
```suggestion
      dead_line_margin: 1.0
      hold_stop_margin_distance: 0.0
```

File: autoware_launch/config/planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/pull_over/pull_over.param.yaml [hunk 8]
```diff
@@ -17,9 +17,6 @@
       backward_goal_search_length: 20.0
       goal_search_interval: 2.0
       longitudinal_margin: 3.0
```
Comment (TakaHoribe) [inline autoware_launch/config/planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/pull_over/pull_over.param.yaml:19]:
```suggestion
      longitudinal_margin: 3.0
      max_lateral_offset: 0.5
      lateral_offset_interval: 0.25
      ignore_distance_from_lane_start: 15.0
```

File: autoware_launch/config/planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/pull_out/pull_out.param.yaml [hunk 9]
```diff
@@ -5,18 +5,18 @@
...
       enable_shift_pull_out: true
       shift_pull_out_velocity: 2.0
       pull_out_sampling_num: 4
+      before_pull_out_straight_distance: 0.0
       minimum_shift_pull_out_distance: 20.0
       maximum_lateral_jerk: 2.0
       minimum_lateral_jerk: 0.5
       deceleration_interval: 15.0
       # geometric pull out
       enable_geometric_pull_out: true
```
Comment (TakaHoribe) [inline autoware_launch/config/planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/pull_out/pull_out.param.yaml:19]:
```suggestion
      enable_geometric_pull_out: true
      divide_pull_out_path: false
```

**Review by TakaHoribe — COMMENTED** [review]
Should we add missing parameters?

**Review by TakaHoribe — APPROVED** [review]

**Pull request merged**